internal/ethapi/api.go

```diff
@@ -882,6 +882,23 @@ func (s *PublicBlockChainAPI) Call(ctx context.Context, args CallArgs, blockNrOr
 	return result.Return(), nil
 }
 
+type estimateGasError struct {
```

Comment:
I think this error type should be exported, s.th. users can use it to cast an error into it and read the fields

Reply:
These are internal packages, they can't be imported either way.

Reply:
I know, but I think this error should be moved to an external package (abi or bind) s.th. users can read the revert reason without having to parse the error string

Reply:
This error is mostly for the API(as Felix required, `ErrorCode` should also be implemented). The internal shouldn't this function directly since it's for API.